src/core/dash_internal.h

```diff
@@ -1506,46 +1506,41 @@ template <bool UV>
 std::enable_if_t<UV, unsigned> Segment<Key, Value, Policy>::CVCOnBump(uint64_t ver_threshold,
                                                                       unsigned bid, unsigned slot,
                                                                       Hash_t hash,
-                                                                      uint8_t result_bid[2]) const {
+                                                                      uint8_t result_bid[3]) const {
   if (bid < kRegularBucketCnt) {
-    // right now we do not migrate entries from nid to bid, only from stash to normal buckets.
+    // Right now we do not migrate entries from nid to bid, only from stash to normal buckets.
+    // The reason for this is that CVCBumpUp implementation swaps the slots of the same bucket
+    // so there is no further action needed.
     return 0;
   }
 
   // Stash case.
+  // There are three actors (interesting buckets). The stash bucket, the target bucket and its
+  // adjacent bucket (probe). To understand the code below consider the cases in CVCBumpUp:
+  // 1. If the bid is not a stash bucket, then just swap the slots of the target.
+  // 2. If there is empty space in target or probe bucket insert the slot there and remove
+  //    it from the stash bucket.
+  // 3. If there is no empty space then we need swap slots with either the target or the probe
+  //    bucket. Furthermore, we might clear the stash bucket so in total all the 3 buckets
```

Comment:
Looks good!!!
`Furthermore, we might clear the stash bucket` - why we might clear the stash bucket?

Reply:
Because we call `target.UnsetStashPtr` which internally calls `ClearStash`

Reply:
ok, ClearStash clears a single entry inside a stash bucket, so maybe `we might remove an entry from the stash bucket, so...`

Reply:
well, technically we do not remove an entry(slot) but clear the stash pointer of the bucket we are about to swap (since insertions after displacement end up on the stash buckets). I slightly reworded.